File: listas/src/main/java/com/victorlh/android/framework/listas/adapter/filtros/FiltroBuscador.java
package com.victorlh.android.framework.listas.adapter.filtros;

import com.victorlh.android.framework.listas.filtros.ItemListaBuscador;
import com.victorlh.tools.ToolsValidacion;

public class FiltroBuscador<T extends ItemListaBuscador> implements FiltroLista<T> {

	private String text;

	public void setText(String text) {
		text = text == null ? "" : text;
		this.text = text.toUpperCase();
	}

	@Override
	public boolean isDentroFiltro(T elemento) {
		String textoBuscador = elemento.getTextoBuscador();
		if (ToolsValidacion.isCadenaNoVacia(textoBuscador)) {
			String upperCase = textoBuscador.toUpperCase();
			return upperCase.contains(text);
		}
		return false;
	}
}

File: listas/src/main/java/com/victorlh/android/framework/listas/adapter/filtros/FiltroBuscador.kt
package com.victorlh.android.framework.listas.adapter.filtros

import com.victorlh.tools.ToolsValidacion
import java.util.*

open class FiltroBuscador<T : ItemListaBuscador> : FiltroLista<T> {

    private var text: String? = null

    fun setText(text: String) {
        this.text = text.uppercase(Locale.getDefault())
    }

    override fun isDentroFiltro(elemento: T): Boolean {
        val textoBuscador = elemento.textoBuscador
        if (ToolsValidacion.isCadenaNoVacia(textoBuscador)) {
            val upperCase = textoBuscador.uppercase(Locale.getDefault())
            return upperCase.contains(text!!)
        }
        return false
    }
}